### PR TITLE
Coverage: Missing test for function length and parameter destructuring. Closes gh-2895

### PR DESCRIPTION
**Pull request by rwaldron**

### CODE REVIEW

File: test/language/destructuring/binding/syntax/destructuring-array-parameters-function-arguments-length.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// Copyright (C) 2020 Rick Waldron. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-destructuring-binding-patterns
+description: >
+  Funciton.length when ArrayBindingPattern in FormalParameterList
```
Comment (leobalter) [inline test/language/destructuring/binding/syntax/destructuring-array-parameters-function-arguments-length.js:7]: typo

File: test/language/destructuring/binding/syntax/destructuring-object-parameters-function-arguments-length.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+// Copyright (C) 2020 Rick Waldron. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-destructuring-binding-patterns
+description: >
+  Funciton.length when ObjectBindingPattern in FormalParameterList
```
Comment (leobalter) [inline test/language/destructuring/binding/syntax/destructuring-object-parameters-function-arguments-length.js:7]: typo

**Review by leobalter — CHANGES_REQUESTED** [review]

**Comment by leobalter** [comment]
Great catch! I thought this would be `length === 0` because of the binding pattern but I'm wrong.

```
# https://tc39.es/ecma262/#sec-function-definitions-static-semantics-expectedargumentcount
Static Semantics: ExpectedArgumentCount

  FormalParameterList : FormalParameter

  1. If HasInitializer of FormalParameter is true, return 0.
  2. Return 1.

# https://tc39.es/ecma262/#sec-destructuring-binding-patterns-static-semantics-hasinitializer
Static Semantics: HasInitializer

  BindingElement : BindingPattern

  1. Return false.
```

**Review by leobalter — APPROVED** [review]

**Comment by rwaldron** [comment]
@leobalter sorry, I should've included better metadata. I'm going to add what you pasted

**Pull request merged**